### PR TITLE
Optimise x space

### PR DESCRIPTION
**Pull request by Delfil**
Optimising space between nodes, making empty screens less common. The spacing also takes interestingness in account.

### CODE REVIEW

File: src/main/java/tudelft/ti2806/pl3/data/gene/GeneData.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public static GeneData parseGenes(String filename) throws IOException {
 	 */
 	protected static void parseGene(String line, ArrayList<Gene> genes, Map<Integer, Gene> geneStart,
 			Map<Integer, Gene> geneEnd, Map<String, Label> geneMap) {
-		String[] tokens = line.split("\t");
+		String[] tokens = line.split("	");
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/data/gene/GeneData.java]: u crazy?

File: src/main/java/tudelft/ti2806/pl3/data/wrapper/operation/GetFirstCombineWrapper.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package tudelft.ti2806.pl3.data.wrapper.operation;
+
+import tudelft.ti2806.pl3.data.wrapper.CombineWrapper;
+import tudelft.ti2806.pl3.data.wrapper.HorizontalWrapper;
+import tudelft.ti2806.pl3.data.wrapper.SpaceWrapper;
+import tudelft.ti2806.pl3.data.wrapper.VerticalWrapper;
+import tudelft.ti2806.pl3.data.wrapper.Wrapper;
+
+public class GetFirstCombineWrapper extends WrapperOperation {
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/data/wrapper/operation/GetFirstCombineWrapper.java]: Remove class?

File: src/main/java/tudelft/ti2806/pl3/data/wrapper/operation/collapse/CalculateCollapseOnSpace.java [hunk 3]
```diff
@@ -1,100 +1,73 @@
...
-			if (left == 0) {
-				count--;
-			} else {
-				avg += left;
+	static float getMinSpaceLeft(HorizontalWrapper wrapper) {
+		return getMinDistance(wrapper.getFirst(), wrapper.getLast());
+	}
+	
+	static float getCompensatedMinSpaceLeft(SpaceWrapper wrapper) {
+		return 1.1f * ((wrapper.getLast().getX() - wrapper.getFirst().getX()) / (wrapper.getLast()
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/data/wrapper/operation/collapse/CalculateCollapseOnSpace.java]: magic number?

File: src/main/java/tudelft/ti2806/pl3/menubar/MenuBarController.java [hunk 4]
```diff
@@ -62,13 +62,13 @@
...
 			+ "All the controls that can be used are listed below. \n"
 			+ "\n"
-			+ "Zooming in     \t+ \n"
-			+ "Zooming out    \t - \n"
-			+ "Reset the view \t R \n"
-			+ "Move the view to the left \t \u2190 \n"
-			+ "Move the view to the right \t \u2192 \n"
-			+ "Gene navigation window \t G \n"
-			+ "Hide/show phylogenetic tree window \t spacebar \n"
+			+ "Zooming in     	+ \n"
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/menubar/MenuBarController.java]: Please no do this

File: src/test/java/tudelft/ti2806/pl3/data/gene/GeneDataTest.java [hunk 5]
```diff
@@ -63,7 +63,7 @@ public void parseSingleGeneTest() {
 		Map<Integer, Gene> geneStart = new HashMap<>();
 		Map<Integer, Gene> geneEnd = new HashMap<>();
 
-		String line = "585\tCorndogp005\tchr\t-\t1110\t1581\t1110\t1581\t1\t1110,\t1581,\t5\tnone\tcmpl\tcmpl\t0,";
+		String line = "585	Corndogp005	chr	-	1110	1581	1110	1581	1	1110,	1581,	5	none	cmpl	cmpl	0,";
```
Comment (borismattijssen) [inline src/test/java/tudelft/ti2806/pl3/data/gene/GeneDataTest.java]: please no do this

**Comment by borismattijssen** [comment]
Please fix. The rest is fine.

File: src/main/java/tudelft/ti2806/pl3/data/wrapper/operation/GetFirstCombineWrapper.java (hunk 2 above)
Comment (Delfil) [inline src/main/java/tudelft/ti2806/pl3/data/wrapper/operation/GetFirstCombineWrapper.java]: Is used in ZoomedGraphModel

**Comment by kwendel** [comment]
lookds good, please merge with master so we can add this new functionality

**Pull request merged**